Sources/Datadog/FeaturesIntegration/TracingToLoggingOutput.swift

```diff
@@ -0,0 +1,44 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// Bridges logs created by Tracing feature to Logging feature's output. This stands for the thin integration layer
+/// between Tracing and Logging features.
+internal struct TracingToLoggingOutput {
+    /// Open Tracing standard log fields:
+    /// https://github.com/opentracing/specification/blob/master/semantic_conventions.md#log-fields-table
+    struct OpenTracingFields {
+        static let message = "message"
+        // TODO: RUMM-477 Support all standard OT log fields and expose them to the user
+    }
+
+    /// Datadog reserved log fields.
+    struct DatadogFields {
+        // TODO: RUMM-478 Add tracing log attributes to the list of reserved log attributes in `LogSanitizer`
+        static let traceID = "dd.trace_id"
+        static let spanID = "dd.span_id"
+    }
+
+    struct DefaultFieldValues {
+        static let message = "Span event"
+    }
+
+    // MARK: - TraceLogOutput
+
+    /// `LogOutput` provided by the `Logging` feature.
+    let loggingOutput: LogOutput
```

Comment:
I was struggling a lot to make up the name for this class. And indeed, it's the adapter pattern which we implement here 👌. `TracingToLoggingAdapter` may sound fine (we bridge features - so "tracing", not "traces").

However, I think this part of `DDTracer` is very clear and readable:
```swift
class DDTracer: Tracer {
    /// Writes `Span` objects to output.
     internal let spanOutput: SpanOutput
     /// Writes span logs to output.
     internal let logOutput: TracingToLoggingOutput

     // ...
}
```
as it clearly says there are separate outputs for `spans` and `logs`. So, how about doing this:
```swift
/// Writes `Span` objects to output.
internal let spanOutput: SpanOutput
/// Writes span logs to output.
internal let logOutput: TracingToLoggingAdapter.LogOutput
```
to keep both a verbose class name and clear usage in local scope?

This could also set up a nice convention for other features integration (i.e. `RUMToTracingAdapter` and `TracingToRUMAdapter` which are coming very soon).

Reply:
it looks good 👌 but i didn't understand how that'll work 🤔 
`Tracer` won't keep an `adapter` instance, it will only keep its `adapter.logOutput`?

or we can just rename it to `TracingLogger` and change it to
```suggestion
    private let loggingOutput: LogOutput
```
so that it will be obvious to use this type as a `Logger`, its private `loggingOutput` will be its implementation detail

```swift
let tracingLogger = TracingLogger(logOutput: realLogOutput)
tracingLogger.writeLogWith(...)
```

i don't have a strong opinion here, wdyt?

Reply:
Keeping it `internal` is a good trade-off for tests in`DDTracerConfigurationTests.swift`, where we make the assertions on `LogBuilder` (`loggingOutput.builder`) to ensure it properly configures the `Logs` that it creates.

> it looks good 👌 but i didn't understand how that'll work 🤔

I mean the `TracingToLoggingAdapter` may become an adapter depending on the `LoggingFeature`:
```swift
internal class TracingToLoggingAdapter {
   let loggingOutput: LogOutput

   init(loggingFeature: LoggingFeature) {
      loggingOutput = LogOutput(fileWriter: loggingFeature.storage.fileWriter)
   }

   struct LogOutput {
      init(fileWriter: FileWriter) { ... }
      // everything from the existing `TracingToLoggingOutput`
   }
}
```

Then, in `TracingFeature`, instead of:
```swift
// MARK: - Integration With Other Features

let loggingFeatureStorage: LoggingFeature.Storage
```
we will store only the logging-agnostic adapter:
```swift
// MARK: - Integration With Other Features

let loggingFeatureAdapter: TracingToLoggingAdapter
```
This will remove the dependency of `LoggingFeature` from `TracingFeature`, making the features isolated again ✨.

Reply:
sounds good 👍

Reply:
Fixed by https://github.com/DataDog/dd-sdk-ios/pull/120/commits/c2ca4409664a1ab99b3be4e27aeac9ffe9f91b63 in #120, as fixing it here will lead to rebasing hell (due to many open PRs).